src/views/homeView/homeViewClass.js

```diff
@@ -1,52 +1,109 @@
-                        const carouselPop = new carousel('Pop', mov.moviesCompilation[0].movies, 3);
-                        const carouselTop = new carousel('Top', mov.moviesCompilation[1].movies, 3);
-                        const carouselFam = new carousel('Fam', mov.moviesCompilation[2].movies, 2);
-                        const footer = new FooterClass();
-
-                        root.innerHTML = homeViewTemplate({
-                            header: header.render(),
-                            mainMovie: mainMovie.render(),
-                            carouselPop: carouselPop.render(),
-                            carouselTop: carouselTop.render(),
-                            carouselFam: carouselFam.render(),
-                            footer: footer.render(),
-                        });
-                        handlerLink();
-                        carouselPop.setHandler();
-                        carouselTop.setHandler();
-                        carouselFam.setHandler();
-                    })
-                    .catch((err) => {
-                        console.error(err);
-                    });
-            })
-            .catch((err) => {
-                console.error(err);
+import homeViewTemplate from './homeView.pug';
+import HeaderClass from 'Components/header/headerClass.js';
+import MainMovieClass from 'Components/mainMovie/mainMovieClass.js';
+import Carousel from 'Components/carousel/carouselClass.js';
+import FooterClass from 'Components/footer/footerClass.js';
+import handlerLink from 'Utils/handlerLink.js';
+import router from 'Routing/router.js';
+import BaseViewClass from '../baseView/baseViewClass.js';
+import { routes } from "Routing/constRouting";
+import LoaderViewClass from "../loaderView/loaderViewClass.js";
+import UserModel from "../../models/User.js"
+import MovieModel from "../../models/Movie.js"
+import MovieCompilationModel from "../../models/MovieCompilation"
+
+import '../../css/home.scss';
+
+export default class HomeViewClass extends BaseViewClass {
+    #user;
+    #mainMovie;
+    #movieCompilations = [];
+
+    async render() {
+        try {
+            const loader = new LoaderViewClass();
+            loader.render();
+
+            const {isAuth, userBody} = await UserModel.auth();
+
+            if (!isAuth) {
+                router.go(routes.LOGIN_VIEW);
+                return;
+            }
+
+            const userData = await Promise.resolve(userBody);
+            this.#user = new UserModel(userData.user);
+
+            const {movBody} = await MovieModel.mainMov();
+            const mainMovieData = await Promise.resolve(movBody);
+            this.#mainMovie = new MovieModel(mainMovieData);
+
+            const {movCompBody} = await MovieCompilationModel.getMovieCompilations();
+            const movieCompilationsData = await Promise.resolve(movCompBody);
+            movieCompilationsData.forEach((movieCompilationData) => {
```

Comment:
`this.#movieCompilations = movieCompilationsData.map((movieCompilationData) => new MovieCompilationModel(movieCompilationData))`